### ISSUE TITLE
Add a debug mode option

### ISSUE THREAD

**Issue by eddiezane** [defenseunicorns/uds-k3d#73]
### Is your feature request related to a problem? Please describe.

While attempting to debug something in Pepr+UDS, I needed to pass some extra k3s args to enable debug logging and such. An idea from @jeff-mccoy is to add a "debug mode".

### Describe the solution you'd like

An arg (or some other control) I can set to enable extra flags for k3s. i.e.

```
--set K3D_DEBUG=true
```

The flags I used in this scenario were the following but we could explore others to cut down on the flag passing from the CLI.

```
uds deploy k3d-core-demo:0.20.0 --set K3D_EXTRA_ARGS="--k3s-arg --kubelet-arg=container-log-max-files=10@server:0 --k3s-arg --kubelet-arg=container-log-max-size=100Mi@server:0 --k3s-arg --debug@server:0"
```

**Comment by eddiezane** [comment]
```--set K3D_EXTRA_ARGS='--k3s-arg --kube-apiserver-arg=--min-request-timeout=120@server:*```

**Comment by eddiezane** [comment]
https://docs.k3s.io/security/hardening-guide#api-server-audit-configuration

**Comment by zachariahmiller** [comment]
@eddiezane This is currently already possible https://github.com/defenseunicorns/uds-k3d/blob/bec028e4077c1f531ef26db5424a30d5253653d7/zarf.yaml#L22
https://github.com/defenseunicorns/uds-k3d?tab=readme-ov-file#additional-info

**Comment by eddiezane** [comment]
@zachariahmiller ah sorry for not being more clear. The ask here was to specifically add a meta option to enable all the debug flags you might need when attempting to debug issues such as turning on api server logs, etc so you don't have to look up all the flags and pass them manually.

Does that make sense?

**Comment by zachariahmiller** [comment]
@eddiezane Yeah that makes sense. I was on the fence when reading the issue for some reason as to whether that was your ask vs just the ability to provide options. 😄
